api/v3/Mailing.php

```diff
@@ -66,10 +66,19 @@ function civicrm_api3_mailing_create($params) {
   else {
     $safeParams = $params;
   }
-  $safeParams['_evil_bao_validator_'] = 'CRM_Mailing_BAO_Mailing::checkSendable';
-  $result = _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $safeParams);
-  return _civicrm_api3_mailing_get_formatResult($result);
-
+  $timestampCheck = TRUE;
+  if (!empty($params['id']) && !empty($params['modified_date'])) {
+    $timestampCheck = _civicrm_api3_compare_timestamps($safeParams['modified_date'], $safeParams['id'], 'Mailing');
+    unset($safeParams['modified_date']);
+  }
+  if (!$timestampCheck) {
+    throw new API_Exception("Mailing has not been saved, Content maybe out of date, please refresh the page and try again");
+  }
+  else {
+    $safeParams['_evil_bao_validator_'] = 'CRM_Mailing_BAO_Mailing::checkSendable';
+    $result = _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $safeParams);
+    return _civicrm_api3_mailing_get_formatResult($result);
+  }
```

Comment:
Non-blocking style nitpick: the last three lines are nested a level deeper (inside the `else {... }` block). This is unnecessary.

Consider how the code would look if we added another validation step stuck with strict `else{}` style. You might get:

```
if (! valid criterion 1) {
  throw new Exception("failed criterion 1");
} else {
  if (! valid criterion 2) {
    throw new Exception("failed criterion 2");
  }
  else {
    doTheNormalThing();
    doTheOtherNormalThing();
  }
}
```

IMHO, the code is more malleable (and a bit easier to read) if you just think of exceptions as exceptional -- the exceptional condition can disrupt flow, but the flow otherwise continues normally, eg

```
if (! valid criterion 1) {
  throw new Exception("failed criterion 1");
}
if (! valid criterion 2) {
  throw new Exception("failed criterion 2");
}

doTheNormalThing();
doTheOtherNormalThing();
```

In the second case, it's "malleable" because you're free to rearrange the validation/logic steps separately -- without munging with the SCM records.